tinyiiod.c

```diff
@@ -67,33 +67,24 @@ char tinyiiod_read_char(struct tinyiiod *iiod)
 	return c;
 }
 
-void tinyiiod_read(struct tinyiiod *iiod, char *buf, size_t len)
+int32_t tinyiiod_read(struct tinyiiod *iiod, char *buf, size_t len)
 {
-	iiod->ops->read(buf, len);
+	return iiod->ops->read(buf, len);
 }
 
 int32_t tinyiiod_read_line(struct tinyiiod *iiod, char *buf, size_t len)
```

Comment:
do we need this wrapper anymore ?

Reply:
this will be removed in a later pull request.

Reply:
ack

Reply:
this `tinyiiod_read_line()` conversion is a bit big
and it should be it's own commit;

though, i wouldn't remove it;

i would do:
```
if (iiod->ops->read_line)
    return iiod->ops->read_line(buf, len);
```

and let the other code remain;
if users want to use their own read_line callback, let's let them; if not, we can fallback to the old one

Reply:
This is a good idea. If the user doesn't provide it's own implementation we can use this one, that is not so efficient, but it is something.

Reply:
looking at how serial_read_line() & network_read_line() are implemented, the  `serial_read()` & `network_read()` callbacks could be used as well;

so, i am un-decided whether to expose `read_line` as an ops

Reply:
This will cause only problems for the layer that uses only serial_read(), since will need to read characters one by one, and check for '\n' or to read chunks of data search for '\n' that will result in even more complications. It is possible to find a '\n' in the middle of the buffer, or maybe you won't find any. So, I would use read_line.